### PR TITLE
LPD-41538 Product Configurations REST Layer

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/LPD-41538

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java [hunk 1]
```diff
@@ -105,36 +298,349 @@ public void testPatchProductIdConfiguration() throws Exception {
...
+				_masterCPConfigurationList.getCPConfigurationListId(),
+				productConfiguration);
+	}
+
+	@Override
+	protected ProductConfiguration
+			testDeleteProductConfigurationByExternalReferenceCode_addProductConfiguration()
+		throws Exception {
+
+		ProductConfiguration productConfiguration =
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java:412]: inline

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java [hunk 2]
```diff
@@ -105,36 +298,349 @@ public void testPatchProductIdConfiguration() throws Exception {
...
+				_masterCPConfigurationList.getCPConfigurationListId(),
+				productConfiguration);
+	}
+
+	@Override
+	protected ProductConfiguration
+			testGetProductConfigurationByExternalReferenceCode_addProductConfiguration()
+		throws Exception {
+
+		ProductConfiguration productConfiguration =
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java:465]: inline

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java [hunk 3]
```diff
@@ -105,36 +298,349 @@ public void testPatchProductIdConfiguration() throws Exception {
...
+
+		return cProduct.getCProductId();
+	}
+
+	@Override
+	protected ProductConfiguration
+			testGraphQLProductConfiguration_addProductConfiguration()
+		throws Exception {
+
+		ProductConfiguration productConfiguration =
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java:564]: inline

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java [hunk 4]
```diff
@@ -105,36 +298,349 @@ public void testPatchProductIdConfiguration() throws Exception {
...
+				_masterCPConfigurationList.getCPConfigurationListId(),
+				productConfiguration);
+	}
+
+	@Override
+	protected ProductConfiguration
+			testPatchProductConfiguration_addProductConfiguration()
+		throws Exception {
+
+		ProductConfiguration productConfiguration =
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java:578]: inline

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java [hunk 5]
```diff
@@ -105,36 +298,349 @@ public void testPatchProductIdConfiguration() throws Exception {
...
+				_masterCPConfigurationList.getCPConfigurationListId(),
+				productConfiguration);
+	}
+
+	@Override
+	protected ProductConfiguration
+			testPatchProductConfigurationByExternalReferenceCode_addProductConfiguration()
+		throws Exception {
+
+		ProductConfiguration productConfiguration =
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductConfigurationResourceTest.java:592]: inline

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Pull request closed without merging**